app/vmselect/prometheus/prometheus.go

```diff
@@ -609,7 +609,14 @@ func SeriesHandler(qt *querytracer.Tracer, startTime time.Time, w http.ResponseW
 	if cp.start == 0 {
 		cp.start = cp.end - defaultStep
 	}
-	sq := storage.NewSearchQuery(cp.start, cp.end, cp.filterss, *maxSeriesLimit)
+	maxSeriesLimit := *maxSeriesLimit
+	if cp.limit > maxSeriesLimit {
```

Comment:
Its seems, that this check also isn't needed.

Reply:
logic updated